### DISCUSSION TITLE
How to limit max fanout/fanin in synthesis stage with OpenROAD-flow-scripts?

### DISCUSSION THREAD

**Discussion by dearmrlv** [The-OpenROAD-Project/OpenROAD-flow-scripts#1773] [Q&A]
I've checked yosys documentation for tcl command: synth, and I cannot figure out options to limit max fanout/fanin in synthesis. Is it possible to limit max fanout/fanin in synthesis stage with OpenROAD-flow-scripts? How to do it?
Thanks for any suggestions!

**Comment by maliberty** [comment]
I don't know of a direct way though you might ask the question in https://github.com/YosysHQ/yosys/discussions or their Slack channel.  I do believe that the .lib's max cap is obeyed so there is an indirect limit.

**Comment by dearmrlv** [comment]
https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/882dc81984dc0a84dded5a1e810c09833aa1e2fb/flow/designs/gf12/coyote/constraint_hier.sdc#L2116
https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/882dc81984dc0a84dded5a1e810c09833aa1e2fb/flow/platforms/gf180/config.mk#L54
https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/882dc81984dc0a84dded5a1e810c09833aa1e2fb/flow/platforms/sky130hd/lib/sky130_dummy_io.lib#L21

Hi, Here are what I found about the max fanout issue in ORFS repository. Can these lines work as they expected? Or these lines just have no effect? @maliberty

**Reply by maliberty** [reply]
You said "in synthesis" and I'm not clear that yosys/abc pick up the SDC constraints.  The .lib may well be effective.  You can always ask on the yosys GH or slack for confirmation.
